Library/Homebrew/cmd/search.rb

```diff
@@ -58,40 +59,67 @@ def search(argv = ARGV)
       query = args.remaining.join(" ")
       string_or_regex = query_regexp(query)
       Descriptions.search(string_or_regex, :desc).print
-    elsif args.remaining.first =~ HOMEBREW_TAP_FORMULA_REGEX
-      query = args.remaining.first
+    else
+      query = args.remaining.join(" ")
+      string_or_regex = query_regexp(query)
 
-      results = begin
-        [Formulary.factory(query).name]
-      rescue FormulaUnavailableError
+      remote_results = if query.match?(HOMEBREW_TAP_FORMULA_REGEX) || query.match?(HOMEBREW_TAP_CASK_REGEX)
         _, _, name = query.split("/", 3)
-        remote_results = search_taps(name)
-        [*remote_results[:formulae], *remote_results[:casks]].sort
+        search_taps(name, silent: true)
+      else
+        search_taps(query, silent: true)
       end
 
-      puts Formatter.columns(results) unless results.empty?
-    else
-      query = args.remaining.join(" ")
-      string_or_regex = query_regexp(query)
-      local_results = search_formulae(string_or_regex)
-      puts Formatter.columns(local_results.sort) unless local_results.empty?
+      local_formulae = if query.match?(HOMEBREW_TAP_FORMULA_REGEX)
+        begin
+          [Formulary.factory(query).name]
+        rescue FormulaUnavailableError
+          []
+        end
+      else
+        search_formulae(string_or_regex)
+      end
 
-      remote_results = search_taps(query)
-      tap_results = [*remote_results[:formulae], *remote_results[:casks]].sort
-      puts Formatter.columns(tap_results) unless tap_results.empty?
+      remote_formulae = remote_results[:formulae]
+      all_formulae = local_formulae + remote_formulae
+
+      local_casks = if !OS.mac?
```

Comment:
Would like to see this use the `extend/os` pattern rather than an `OS.mac?` check.